src/commands/midi.ts

```diff
@@ -36,15 +37,95 @@ function playMidi(contextString: string, editor: TextEditor) {
 	else
 		chordType = lineContext(contextString);
 
+	//turns chordtype name into a note for flag sounds
+	chordType = convertChordType(chordType);
+
+	//changes instrument based on chordtype value
+	instrument = changeMidiInstrument(chordType);
+	output.send([0xC0, instrument]);
+
+	//plays sound for flag
 	output.note(0, chordType, 127, 550);
+
+	//changes sounds back to piano
+	output.send([0xC0, 0]);
+}
+
+function changeMidiInstrument(chordType: string): number
+{
+	var num = 0;
+
+	//Changes midi to piano
+	if(chordType == 'D2' ||chordType == 'D4' || chordType == 'D6')
+		num = 0;
+
+	//Changes midi to violin
+	if(chordType == 'E2' ||chordType == 'E4' || chordType == 'E6')
+		num = 40;
+
+	//Changes midi to guitar
+	if(chordType == 'F2' ||chordType == 'F4' || chordType == 'F6')
+		num = 24;
+
+	//Changes midi to marimba
+	if(chordType == 'G2' ||chordType == 'G4' || chordType == 'G6')
+		num = 12;
+
+	return num;
+}
+
+function convertChordType(chordType: string): string
+{
+	if(chordType === "Piano1")
+		chordType = "D2";
+	if(chordType === "Piano2")
+		chordType = "D4";
+	if(chordType === "Piano3")
+		chordType = "D6";
+
+	if(chordType === "Violin1")
+		chordType = "E2";
+	if(chordType === "Violin2")
+		chordType = "E4";
+	if(chordType === "Violin3")
+		chordType = "E6";
+
+	if(chordType === "Guitar1")
+		chordType = "F2";
+	if(chordType === "Guitar2")
+		chordType = "F4";
+	if(chordType === "Guitar3")
+		chordType = "F6";
+
+	if(chordType === "Marimba1")
+		chordType = "G2";
+	if(chordType === "Marimba2")
+		chordType = "G4";
+	if(chordType === "Marimba3")
+		chordType = "G6";
+
+	return chordType;
 }
 
 //use to play notes when a flag is added. mainly used in audioflags.ts and this file
 export function playFlagMidi(note: string) {
 	if (shouldPlayMIDINote)
 	{
 		var output = jzz().openMidiOut();
+		var instrument = 0;
+
+		//turns chordtype name into a note for flag sounds
+		note = convertChordType(note);
+
+		//changes instrument based on chordtype value
+		instrument = changeMidiInstrument(note);
+		output.send([0xC0, instrument]);
+
+		//plays sound for flag
 		output.note(0, note, 127, 550);
+
+		//changes sounds back to piano
+		output.send([0xC0, 0]);
 	}
 }
```

Comment:
This function can be changed to accept a note and instrument as strings (among other things if you care to add this functionality). The sole purpose of this function should be to accept the parameters and play the sound accordingly. This function could also be renamed to better fit its new use case.